docs-markdown/src/controllers/moniker-controller.ts

```diff
@@ -0,0 +1,90 @@
+"use strict";
+
+import { Range, TextEditor, window } from "vscode";
+import { insertContentToEditor, setCursorPosition, isMarkdownFileCheck, noActiveEditorMessage } from "../helper/common";
+import { isCursorInsideYamlHeader } from "../helper/yaml-metadata";
+import { sendTelemetryData } from "../helper/telemetry";
+
+const telemetryCommand: string = "insertMoniker";
+let sign: string = "";
+
+export function insertMonikerCommand() {
+    const commands = [
+        { command: insertMoniker.name, callback: insertMoniker },
+    ];
+    return commands;
+}
+
+export function insertMoniker() {
+    const editor = window.activeTextEditor;
+    if (!editor) {
+        noActiveEditorMessage();
+        return;
+    }
+
+    if (!isMarkdownFileCheck(editor, false)) {
+        return;
+    }
+
+    //is cursor on a new line 
+    if (isContentOnCurrentLine(editor)) {
+        window.showErrorMessage("Moniker must be inserted on a new line.");
+        return;
+    }
+
+    const moniker_options = [
```

Comment:
We try to keep the variable names as camelCase. Please change `moniker_options` to `monikerOptions`.